### ISSUE TITLE
🐛 `2024.1.0` → CurvePreferences includes unsupported curve

### ISSUE THREAD

**Issue by buroa** [cloudflare/cloudflared#1143]
**Describe the bug**
`Failed to create new quic connection error="failed to dial to edge with quic: INTERNAL_ERROR (local): tls: CurvePreferences includes unsupported curve" connIndex=0 event=0 ip=198.41.200.193`

**To Reproduce**
1. Run `cloudflared`: https://github.com/buroa/k8s-gitops/blob/master/kubernetes/apps/networking/cloudflared/app/helmrelease.yaml#L38-L42

If it's an issue with Cloudflare Tunnel:
1. Tunnel ID: `bfb3d448-8db6-4dda-b84f-11822e0fc62f`
2. cloudflared config: https://github.com/buroa/k8s-gitops/blob/master/kubernetes/apps/networking/cloudflared/app/configs/config.yaml

**Expected behavior**
Doesn't crash.

**Environment and versions**
 - OS: `Linux`
 - Architecture: `AMD64`
 - Version: `2024.1.0`

**Comment by kznrluk** [comment]
related #1132

**Comment by buroa** [comment]
Thanks @kznrluk!

**Comment by kms0219kms** [comment]
2024.1.0 has a serious issue/bug, so it's been unreleased; it will be back soon.

**Comment by yjutard** [comment]
Hi fellow developers,

if you're using brew and are stuck with this faulty version, here is how to unblock:

```bash
## remove faulty version
brew uninstall cloudflared

## update everything else
brew update

## install last known working version
curl https://raw.githubusercontent.com/Homebrew/homebrew-core/08b1d877b616312f844957a35bb37ffdfc849c75/Formula/c/cloudflared.rb --location --output ~/cloudflared.rb
brew install ~/cloudflared.rb
```

Credits to https://stackoverflow.com/a/7787703/587407

**Comment by chungthuang** [comment]
Sorry for the trouble. We have released 2024.1.1 to fix the issue.

**Comment by buroa** [comment]
Thanks @chungthuang :)

**Comment by chbgdn** [comment]
@chungthuang I built 2024.1.1, but bug from this issue is present for me, it not fixed.

**Comment by chbgdn** [comment]
> @chungthuang I built 2024.1.1, but bug from this issue is present for me, it not fixed.

Ok, I found message about new patched go toolchain. After build with it all ok, but I think it need to be referenced in readme.

**Comment by valeriangalliat** [comment]
Still having that issue on the Homebrew MacOS/ARM64 2024.1.1 version

**Comment by kms0219kms** [comment]
+1 same; Still having issue

**Comment by AndreTheHunter** [comment]
+1 same. Create new issue https://github.com/cloudflare/cloudflared/issues/1148
